tests/macro_expansion.cpp

```diff
@@ -0,0 +1,113 @@
+#include "ast/passes/macro_expansion.h"
+#include "ast/passes/parser.h"
+#include "ast/passes/printer.h"
+#include "mocks.h"
+#include "gtest/gtest.h"
+
+namespace bpftrace::test::macro_expansion {
+
+using ::testing::HasSubstr;
+
+void test(const std::string& input,
+          const std::string& error = "",
+          const std::string& warn = "")
+{
+  auto mock_bpftrace = get_mock_bpftrace();
+  BPFtrace& bpftrace = *mock_bpftrace;
+  bpftrace.config_->unstable_macro = true;
+
+  // The input provided here is embedded into an expression.
+  ast::ASTContext ast("stdin", input);
+  std::stringstream msg;
+  msg << "\nInput:\n" << input << "\n\nOutput:\n";
+
+  // N.B. No macro or tracepoint expansion.
```

Comment:
Probably best to clarify this, assuming it is accurate
```suggestion
  // N.B. No C macro or tracepoint expansion.
```